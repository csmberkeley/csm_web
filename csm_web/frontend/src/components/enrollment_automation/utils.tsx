/**
 * Convert 24hr time to 12hr time
 *
 * @param time string in format "HH:MM"
 */
export function formatTime(time: number): string {
  const hours = Math.floor(time / 60);
  const minutes = time % 60;
  const ampm = hours >= 12 ? "PM" : "AM";
  if (minutes == 0) {
    return `${hours > 12 ? hours % 12 : hours} ${ampm}`;
  }
  return `${hours > 12 ? hours % 12 : hours}:${minutes} ${ampm}`;
}

/**
 * Serialize time into HH:MM format
 *
 * @param time number of minutes past midnight
 */
export function serializeTime(time: number): string {
  const hours = Math.floor(time / 60);
  const minutes = time % 60;
  return `${hours < 10 ? "0" : ""}${hours}:${minutes < 10 ? "0" : ""}${minutes}`;
}

/**
 * Parses a 24h time string and returns the number of minutes past midnight.
 * @param time 24h time string
 * @returns minutes past midnight
 */
export function parseTime(time: string): number {
  const [hours, minutes] = time.split(":");
  return parseInt(hours) * 60 + parseInt(minutes);
}
