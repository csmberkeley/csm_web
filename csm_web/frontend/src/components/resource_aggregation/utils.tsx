import { emptyWorksheet, Worksheet, WorksheetKeys } from "./ResourceTypes";

export enum ResourceWorksheetActionType {
  UpdateField,
  DeleteWorksheet,
  DeleteFile,
  AddWorksheet
}

export enum ResourceWorksheetKind {
  EXISTING = "existing",
  NEW = "new"
}

/**
 * Various possible actions that can be submitted to the reducer.
 * Each action can have different parameters, listed as part of the union type.
 *
 * `worksheetId` is either the existing worksheet ID, or the index of the new worksheet in the array.
 */
export type ResourceWorksheetReducerAction =
  | {
      type: ResourceWorksheetActionType.UpdateField;
      kind: ResourceWorksheetKind;
      worksheetId: number;
      field: WorksheetKeys.name;
      value: string;
    }
  | {
      type: ResourceWorksheetActionType.UpdateField;
      kind: ResourceWorksheetKind;
      worksheetId: number;
      field: WorksheetKeys.worksheetFile | WorksheetKeys.solutionFile;
      value: FileList;
    }
  | {
      type: ResourceWorksheetActionType.UpdateField;
      kind: ResourceWorksheetKind;
      worksheetId: number;
      field: WorksheetKeys.worksheetSchedule | WorksheetKeys.solutionSchedule;
      value: string | null;
    }
  | {
      type: ResourceWorksheetActionType.DeleteWorksheet;
      kind: ResourceWorksheetKind;
      worksheetId: number;
    }
  | {
      type: ResourceWorksheetActionType.DeleteFile;
      kind: ResourceWorksheetKind;
      worksheetId: number;
      field: WorksheetKeys.worksheetFile | WorksheetKeys.solutionFile;
    }
  | { type: ResourceWorksheetActionType.AddWorksheet };

interface ResourceWorksheetReducerState {
  existingWorksheets: Map<number, Worksheet>;
  newWorksheets: Worksheet[];
}

/**
 * Reducer to keep track of worksheets, and to handle all possible actions on the worksheets..
 */
export function resourceWorksheetReducer(
  { existingWorksheets, newWorksheets }: ResourceWorksheetReducerState,
  action: ResourceWorksheetReducerAction
): ResourceWorksheetReducerState {
  /**
   * Retrieves a worksheet, and calls the callback function on the worksheet.
   * The callback should modify the worksheet in-place.
   *
   * `worksheetKind` determines which variable the worksheet is taken from.
   * If existing, `worksheetId` is the ID of the worksheet to retrieve;
   * if new, `worksheetId` is the index in the array to retrieve.
   */
  const retrieveAndExecuteWorksheet = (
    worksheetKind: ResourceWorksheetKind,
    worksheetId: number,
    callback: (worksheet: Worksheet) => void
  ): ResourceWorksheetReducerState => {
    // retrieve the worksheet
    let worksheet: Worksheet;
    if (worksheetKind === ResourceWorksheetKind.EXISTING) {
      if (existingWorksheets.has(worksheetId)) {
        worksheet = existingWorksheets.get(worksheetId)!;
      } else {
        throw new Error(`Worksheet not found: id ${worksheetId}`);
      }
    } else {
      worksheet = newWorksheets[worksheetId];
    }

    // modify the worksheet
    callback(worksheet);

    // return updated worksheets
    if (worksheetKind === ResourceWorksheetKind.EXISTING) {
      return { existingWorksheets: new Map(existingWorksheets), newWorksheets };
    } else {
      return { existingWorksheets, newWorksheets: [...newWorksheets] };
    }
  };

  switch (action.type) {
    case ResourceWorksheetActionType.UpdateField: {
      switch (action.field) {
        // Update the worksheet name
        case WorksheetKeys.name: {
          return retrieveAndExecuteWorksheet(action.kind, action.worksheetId, worksheet => {
            worksheet[WorksheetKeys.name] = action.value;
          });
        }

        // Update the worksheet files
        case WorksheetKeys.worksheetFile:
        case WorksheetKeys.solutionFile: {
          return retrieveAndExecuteWorksheet(action.kind, action.worksheetId, worksheet => {
            worksheet[action.field] = action.value[0];
          });
        }

        // Update the worksheet schedules
        case WorksheetKeys.worksheetSchedule:
        case WorksheetKeys.solutionSchedule: {
          return retrieveAndExecuteWorksheet(action.kind, action.worksheetId, worksheet => {
            worksheet[action.field] = action.value;
          });
        }
      }
      break;
    }

    // Delete an existing worksheet
    case ResourceWorksheetActionType.DeleteWorksheet: {
      if (action.kind === ResourceWorksheetKind.EXISTING) {
        // special handling for marking a worksheet as deleted
        return retrieveAndExecuteWorksheet(action.kind, action.worksheetId, worksheet => {
          if (worksheet.deleted == undefined || !(worksheet.deleted instanceof Array)) {
            worksheet.deleted = [];
          }
          if (!worksheet.deleted.includes("worksheet")) {
            worksheet.deleted.push("worksheet");
          }
        });
      } else {
        // since new worksheets aren't registered in the database, we can just splice the worksheet from the array.
        const updatedWorksheets = [...newWorksheets];
        updatedWorksheets.splice(action.worksheetId, 1);
        return { existingWorksheets, newWorksheets: updatedWorksheets };
      }
    }

    // Delete an existing worksheet's file
    case ResourceWorksheetActionType.DeleteFile: {
      if (action.kind === ResourceWorksheetKind.EXISTING) {
        // special handling for marking a worksheet file as deleted
        return retrieveAndExecuteWorksheet(action.kind, action.worksheetId, worksheet => {
          if (worksheet.deleted == undefined || !(worksheet.deleted instanceof Array)) {
            worksheet.deleted = [];
          }
          if (!worksheet.deleted.includes(action.field)) {
            worksheet.deleted.push(action.field);
          }
        });
      } else {
        // since new worksheets aren't reigstered in the database, we can just delete the file from the object.
        const worksheet = newWorksheets[action.worksheetId];
        worksheet[action.field] = "";
        return { existingWorksheets, newWorksheets: [...newWorksheets] };
      }
    }

    case ResourceWorksheetActionType.AddWorksheet: {
      return { existingWorksheets, newWorksheets: [...newWorksheets, emptyWorksheet()] };
    }
  }
}
