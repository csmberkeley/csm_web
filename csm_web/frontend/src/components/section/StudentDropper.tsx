import React, { useState } from "react";
import { useDropStudentMutation } from "../../utils/queries/sections";
import Modal from "../Modal";

interface StudentDropperProps {
  id: number;
  sectionId: number;
  name: string;
}

export default function StudentDropper({ id, sectionId, name }: StudentDropperProps) {
  const [showDropPrompt, setShowDropPrompt] = useState(false);
  const [drop, setDrop] = useState(false);
  const [ban, setBan] = useState(false);

  /**
   * Mutation to drop a student from the section.
   */
  const studentDropMutation = useDropStudentMutation(id, sectionId);

  function handleClickDrop() {
    studentDropMutation.mutate({ banned: ban });
    setShowDropPrompt(false);
  }

  return (
    <span className={`student-dropper ${showDropPrompt ? "ban-prompt-visible" : ""}`}>
      <span title="Drop student from section" className="inline-plus-sign" onClick={() => setShowDropPrompt(true)}>
        Ã—
      </span>
      {showDropPrompt && (
        <Modal closeModal={() => setShowDropPrompt(false)}>
          <div className="studentDropper">
            <div className="studentDropperHeadItem">DROP Student</div>
            <div>
              <input type="checkbox" id="drop" name="drop" onChange={e => setDrop(e.target.checked)} />
              <label className="studentDropperCheckboxLabel" htmlFor="drop">
                I would like to DROP {name} from this section.
              </label>
              <br></br>
            </div>
            <div className="studentDropperHeadItem">BAN Student</div>
            <div>
              <input type="checkbox" id="ban" name="ban" onChange={e => setBan(e.target.checked)} disabled={!drop} />
              <label className="studentDropperCheckboxLabel" htmlFor="ban">
                I would like to BAN {name} from this course.
              </label>
              <br></br>
            </div>
            <div className="studentDropperSubmitWrapper">
              <button
                className="studentDropperSubmit"
                id="dropper submit button"
                onClick={handleClickDrop}
                disabled={!drop}
              >
                Submit
              </button>
            </div>
          </div>
        </Modal>
      )}
    </span>
  );
}
