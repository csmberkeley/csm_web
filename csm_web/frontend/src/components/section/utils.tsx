export const MONTH_NUMBERS = Object.freeze({
  Jan: 1,
  Feb: 2,
  Mar: 3,
  Apr: 4,
  May: 5,
  Jun: 6,
  Jul: 7,
  Aug: 8,
  Sep: 9,
  Oct: 10,
  Nov: 11,
  Dec: 12
});

export const DAYS_OF_WEEK = Object.freeze([
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday",
  "Sunday"
]);

export function formatDate(dateString) {
  /*
   * Example:
   * formatDate("Jan. 6, 2020") --> "1/6"
   */
  const [month, dayAndYear] = dateString.split(".");
  const day = dayAndYear.split(",")[0].trim();
  return `${MONTH_NUMBERS[month]}/${day}`;
}

export function dateSort(date1, date2) {
  const [month1, day1] = formatDate(date1)
    .split("/")
    .map(part => Number(part));
  const [month2, day2] = formatDate(date2)
    .split("/")
    .map(part => Number(part));
  return month2 - month1 || day2 - day1;
}

export function zeroPadTwoDigit(num) {
  return num < 10 ? `0${num}` : num;
}
