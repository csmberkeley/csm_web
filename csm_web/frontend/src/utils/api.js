import Cookies from "js-cookie";

/* Why is this useful? Trying to access a mispelled property of an object will result in an error
   during build, whereas a mispelled string literal would cause an error only when fetch
   was actually called */
const HTTP_METHODS = Object.freeze({
  POST: "POST",
  GET: "GET",
  PUT: "PUT",
  PATCH: "PATCH",
  DELETE: "DELETE"
});
export { HTTP_METHODS };

export function fetchWithMethod(endpoint, method, data = {}) {
  if (!HTTP_METHODS.hasOwnProperty(method)) {
    // check that method choice is valid
    throw new Error(
      "HTTP method must be one of: POST, GET, PUT, PATCH, or DELETE"
    );
  }
  return fetch(`/api/${endpoint}`, {
    method: method,
    credentials: "same-origin",
    headers: {
      "X-CSRFToken": Cookies.get("csrftoken"),
      Accept: "application/json",
      "Content-Type": "application/json"
    },
    body: JSON.stringify(data)
  });
}

export function fetchJSON(endpoint) {
  return fetch(`/api/${endpoint}`).then(response => response.json());
}
