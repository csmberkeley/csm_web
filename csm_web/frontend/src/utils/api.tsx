import Cookies from "js-cookie";

/* Why is this useful? Trying to access a mispelled property of an object will result in an error
   during build, whereas a mispelled string literal would cause an error only when fetch
   was actually called */
export const HTTP_METHODS = Object.freeze({
  POST: "POST",
  GET: "GET",
  PUT: "PUT",
  PATCH: "PATCH",
  DELETE: "DELETE"
});

export function normalizeEndpoint(endpoint: string) {
  if (endpoint[0] == "/") {
    endpoint = endpoint.slice(1);
  }
  if (endpoint[endpoint.length - 1] != "/") {
    endpoint += "/";
  }
  return `/api/${endpoint}`;
}

export function fetchWithMethod(endpoint: string, method: string, data: any = {}, isFormData = false) {
  if (!Object.prototype.hasOwnProperty.call(HTTP_METHODS, method)) {
    // check that method choice is valid
    throw new Error("HTTP method must be one of: POST, GET, PUT, PATCH, or DELETE");
  }
  if (isFormData) {
    return fetch(normalizeEndpoint(endpoint), {
      method: method,
      credentials: "same-origin",
      headers: {
        "X-CSRFToken": Cookies.get("csrftoken")
      },
      body: data
    });
  }
  return fetch(normalizeEndpoint(endpoint), {
    method: method,
    credentials: "same-origin",
    headers: {
      "X-CSRFToken": Cookies.get("csrftoken"),
      Accept: "application/json",
      "Content-Type": "application/json"
    },
    body: JSON.stringify(data)
  });
}

export async function fetchJSON(endpoint: string) {
  const response = await fetch(normalizeEndpoint(endpoint), { credentials: "same-origin" });
  return await response.json();
}
