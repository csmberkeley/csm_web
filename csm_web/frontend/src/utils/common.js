import React from "react";

export function alert_modal(message, callback) {
  UIkit.modal.alert(message).then(callback);
}

export function Modal(props) {
  return (
    <div id={props.id} data-uk-modal>
      <div className="uk-modal-dialog uk-modal-body">
        <button
          className="uk-modal-close"
          type="button"
          style={{ float: "right" }}
          data-uk-icon="icon: close"
        >
          {" "}
        </button>
        <h2 className="uk-modal-title" style={{ marginTop: "0px" }}>
          {props.title}
        </h2>
        {props.children}
      </div>
    </div>
  );
}
