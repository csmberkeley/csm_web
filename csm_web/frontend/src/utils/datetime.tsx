import { DateTime, DateTimeFormatOptions, IANAZone, Interval } from "luxon";

export const DEFAULT_TIMEZONE = IANAZone.create("America/Los_Angeles");

export const DEFAULT_LONG_LOCALE_OPTIONS: DateTimeFormatOptions = {
  weekday: "long",
  month: "numeric",
  day: "numeric",
  hour: "numeric",
  minute: "numeric",
  hour12: true,
  timeZoneName: "short"
};

export const DATETIME_INITIAL_INVALID = DateTime.invalid("initial value");
export const INTERVAL_INITIAL_INVALID = Interval.invalid("initial value");

/**
 * Format a date for display.
 *
 * @param datetime - datetime object to format
 * @returns formatted date
 */
export const formatDate = (datetime: DateTime): string => {
  return datetime.toLocaleString(DateTime.DATE_SHORT);
};

/**
 * Format datetime as "HH:MM AM/PM".
 *
 * If the minutes are 0, displays "HH AM/PM" if 12-hour.
 * 24-hour time display has no special handling.
 *
 * @param number representing the number of minutes past 12:00 AM
 */
export function formatTime(datetime: DateTime) {
  if (datetime.get("minute") == 0) {
    return datetime.toLocaleString({ hour: "numeric", hour12: true });
  } else {
    return datetime.toLocaleString({ ...DateTime.TIME_SIMPLE, hour12: true, hour: "numeric" });
  }
}

/**
 * Format an interval object.
 *
 * If the minutes for both endpoints are 0, then minutes are not displayed
 * (ex. 10 - 11 AM, instead of 10:00 - 11:00 AM).
 *
 */
export function formatInterval(interval: Interval): string {
  if (interval.start!.get("minute") != 0 || interval.end!.get("minute") != 0) {
    return interval.toLocaleString({ hour: "numeric", minute: "2-digit", hour12: true });
  } else {
    return interval.toLocaleString({ hour: "numeric", hour12: true });
  }
}
