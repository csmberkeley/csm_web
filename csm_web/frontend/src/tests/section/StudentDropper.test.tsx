import "@testing-library/jest-dom";
import { act, fireEvent, render, waitFor } from "@testing-library/react";
import React from "react";
import StudentDropper from "../../components/section/StudentDropper";
import * as api from "../../utils/api";

// mock modal
jest.mock("../../components/Modal", () => {
  return {
    __esModule: true,
    default: function Modal(props: { closeModal: React.MouseEventHandler<Element>; children: React.ReactNode }) {
      return (
        <div className="_mock-modal">
          <button onClick={props.closeModal}>Close</button>
          {props.children}
        </div>
      );
    }
  };
});

describe("StudentDropper", () => {
  it("should render correctly without interaction", () => {
    const reloadSection = jest.fn();
    const component = render(<StudentDropper id={1} name="John Doe" reloadSection={reloadSection} />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should render modal correctly after clicking x", () => {
    const reloadSection = jest.fn();
    const component = render(<StudentDropper id={1} name="Test Student" reloadSection={reloadSection} />);

    act(() => {
      // click drop button to bring up modal
      fireEvent.click(component.getByTitle("Drop student from section"));
    });

    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should close modal correctly", () => {
    const reloadSection = jest.fn();
    const component = render(<StudentDropper id={1} name="Test Student" reloadSection={reloadSection} />);

    act(() => {
      // click drop button to bring up modal
      fireEvent.click(component.getByTitle("Drop student from section"));
    });

    act(() => {
      // click close button to close modal
      fireEvent.click(component.getByRole("button", { name: /close/i }));
    });

    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should drop student after confirming", async () => {
    // mock fetchWithMethod, checking call arguments
    const spyFetchWithMethod = jest.spyOn(api, "fetchWithMethod");
    spyFetchWithMethod.mockImplementation(async (endpoint: string, method: string, body: any) => {
      // have leeway when checking endpoint
      const normalizedEndpoint = api.normalizeEndpoint(endpoint);
      expect(normalizedEndpoint).toBe("/api/students/1/drop/");
      expect(method).toBe(api.HTTP_METHODS.PATCH);
      expect(body).toEqual({ banned: false });

      return null as any;
    });

    const reloadSection = jest.fn();
    const component = render(<StudentDropper id={1} name="Test Student" reloadSection={reloadSection} />);

    // click drop button to bring up modal
    act(() => {
      fireEvent.click(component.getByTitle("Drop student from section"));
    });

    // submit button should be disabled
    const confirmButton = component.getByText("Submit");
    expect(confirmButton).toBeDisabled();
    // ban button should be disabled
    const banCheckbox = component.getByRole("checkbox", { name: /ban/i });
    expect(banCheckbox).toBeDisabled();
    // drop checkbox should be unchecked
    const dropCheckbox = component.getByRole("checkbox", { name: /drop/i });
    expect(dropCheckbox).not.toBeChecked();

    // check drop checkbox
    act(() => {
      fireEvent.click(dropCheckbox);
    });

    // drop checkbox should now be checked
    expect(dropCheckbox).toBeChecked();
    // submit button should be enabled
    expect(confirmButton).toBeEnabled();
    // ban button should be enabled
    expect(banCheckbox).toBeEnabled();
    // ban button should be unchecked
    expect(banCheckbox).not.toBeChecked();

    // click confirm button to drop student
    act(() => {
      fireEvent.click(confirmButton);
    });

    // wait for fetch to finish
    waitFor(() => {
      expect(spyFetchWithMethod).toHaveBeenCalled();
      expect(reloadSection).toHaveBeenCalled();
    });
  });

  it("should drop and ban student after confirming", async () => {
    // mock fetchWithMethod, checking call arguments
    const spyFetchWithMethod = jest.spyOn(api, "fetchWithMethod");
    spyFetchWithMethod.mockImplementation(async (endpoint: string, method: string, body: any) => {
      // have leeway when checking endpoint
      const normalizedEndpoint = api.normalizeEndpoint(endpoint);
      expect(normalizedEndpoint).toBe("/api/students/1/drop/");
      expect(method).toBe(api.HTTP_METHODS.PATCH);
      expect(body).toEqual({ banned: true });

      return null as any;
    });

    const reloadSection = jest.fn();
    const component = render(<StudentDropper id={1} name="Test Student" reloadSection={reloadSection} />);

    // click drop button to bring up modal
    act(() => {
      fireEvent.click(component.getByTitle("Drop student from section"));
    });

    // submit button should be disabled
    const confirmButton = component.getByText("Submit");
    expect(confirmButton).toBeDisabled();
    // ban button should be disabled
    const banCheckbox = component.getByRole("checkbox", { name: /ban/i });
    expect(banCheckbox).toBeDisabled();
    // drop checkbox should be unchecked
    const dropCheckbox = component.getByRole("checkbox", { name: /drop/i });
    expect(dropCheckbox).not.toBeChecked();

    // check drop checkbox
    act(() => {
      fireEvent.click(dropCheckbox);
    });

    // drop checkbox should now be checked
    expect(dropCheckbox).toBeChecked();
    // submit button should be enabled
    expect(confirmButton).toBeEnabled();
    // ban button should be enabled
    expect(banCheckbox).toBeEnabled();
    // ban button should be unchecked
    expect(banCheckbox).not.toBeChecked();

    // check ban checkbox
    act(() => {
      fireEvent.click(banCheckbox);
    });

    // drop checkbox should now be checked
    expect(dropCheckbox).toBeChecked();
    // submit button should be enabled
    expect(confirmButton).toBeEnabled();
    // ban button should be enabled
    expect(banCheckbox).toBeEnabled();
    // ban button should be checked
    expect(banCheckbox).toBeChecked();

    // click confirm button to drop student
    act(() => {
      fireEvent.click(confirmButton);
    });

    // wait for fetch to finish
    waitFor(() => {
      expect(spyFetchWithMethod).toHaveBeenCalled();
      expect(reloadSection).toHaveBeenCalled();
    });
  });
});
