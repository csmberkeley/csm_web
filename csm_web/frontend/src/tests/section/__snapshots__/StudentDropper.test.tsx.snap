// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentDropper should close modal correctly 1`] = `
<DocumentFragment>
  <span
    class="student-dropper "
  >
    <x-svg
      classname="icon inline-plus-sign"
      title="Drop student from section"
    />
  </span>
</DocumentFragment>
`;

exports[`StudentDropper should render correctly without interaction 1`] = `
<DocumentFragment>
  <span
    class="student-dropper "
  >
    <x-svg
      classname="icon inline-plus-sign"
      title="Drop student from section"
    />
  </span>
</DocumentFragment>
`;

exports[`StudentDropper should render modal correctly after clicking x 1`] = `
<DocumentFragment>
  <span
    class="student-dropper ban-prompt-visible"
  >
    <x-svg
      classname="icon inline-plus-sign"
      title="Drop student from section"
    />
    <div
      class="_mock-modal"
    >
      <button>
        Close
      </button>
      <div>
        <h2
          class="student-dropper-head-item"
        >
          DROP Student
        </h2>
        <div
          class="student-dropper-checkbox-container"
        >
          <input
            id="drop"
            name="drop"
            type="checkbox"
          />
          <label
            class="student-dropper-checkbox-label"
            for="drop"
          >
            I would like to DROP Test Student from this section.
          </label>
          <br />
        </div>
      </div>
      <div>
        <h2
          class="student-dropper-head-item"
        >
          BAN Student
        </h2>
        <div
          class="student-dropper-checkbox-container"
        >
          <input
            disabled=""
            id="ban"
            name="ban"
            type="checkbox"
          />
          <label
            class="student-dropper-checkbox-label"
            for="ban"
          >
            I would like to BAN Test Student from this course.
          </label>
          <br />
        </div>
      </div>
      <div
        class="student-dropper-submit-wrapper"
      >
        <button
          class="danger-btn"
          disabled=""
        >
          Submit
        </button>
      </div>
    </div>
  </span>
</DocumentFragment>
`;
